### PR TITLE
[DROOLS-1491] allow a type declaration to extend a Java class

### PR DESCRIPTION
**Pull request by mariofusco**

### CODE REVIEW

File: drools-compiler/src/test/java/org/drools/compiler/integrationtests/ExtendsTest.java [hunk 1]
```diff
@@ -977,7 +977,7 @@ public void testDeclareInheritance() throws Exception {
     }
 
     @Test
-    public void testDeclareExtendsMissingDeclareForParent() {
+    public void testDeclareExtendsJavaParent() {
```
Comment (mariofusco) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/ExtendsTest.java:980]: @tarilabs This test and the next one mistakenly didn't allow a type declaration to extend a Java class as a fix for this https://issues.jboss.org/browse/DROOLS-289

Comment (tarilabs) [reply drools-compiler/src/test/java/org/drools/compiler/integrationtests/ExtendsTest.java:980]: ok

File: drools-compiler/src/test/java/org/drools/compiler/integrationtests/ExtendsTest.java [hunk 2]
```diff
@@ -1004,10 +1001,22 @@ public void testDeclareExtendsMissingDeclareForParentOuterPackaga() {
...
+        String drl = "package org.drools.test; \n" +
+                     "import org.drools.compiler.integrationtests.ExtendsTest.Y; \n" +
+                     "declare Student extends Y end \n" +
+                     "";
+        KnowledgeBuilder kBuilder = KnowledgeBuilderFactory.newKnowledgeBuilder(  );
+        kBuilder.add( new ByteArrayResource( drl.getBytes() ), ResourceType.DRL );
+        if ( kBuilder.hasErrors() ) {
+            System.err.println( kBuilder.getErrors() );
+        }
+        assertTrue( kBuilder.hasErrors() );
```
Comment (tarilabs) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/ExtendsTest.java:1018]: ok

**Review by tarilabs — APPROVED** [review]
+1

**Comment by baldimir** [comment]
+1

**Pull request merged**